staging/kos/pkg/apis/network/types.go

```diff
@@ -30,6 +31,32 @@ type ExtendedObjectMeta struct {
 	// +listMapKey=section
 	// +optional
 	Writes WriteSet
+
+	// LastClientWrite identifies the latest client write among those needed to
+	// implement the API object. Unlike `Writes`, it might describe a write on
+	// an API object other than the owning API object.
+	// +optional
+	LastClientWrite ClientWrite
```

Comment:
I wonder whether instead of focusing on `Last<relevant-lifecycle-event>` we should record all events and add functions to retrieve the last one. It would be a generalization of `WriteSet` (the preceding field) to store events other than Object writes (such as controller starts and writes on different, but related, API objects), with the catch that some events cannot be recorded because they do not result in a write to the API object. Also, we would probably still want to store events in separate sets on the basis of their "type" because consumers usually know the type of event (e.g. write, controller start, etc...) they care about.

Reply:
I think we should not here invent a framework for tracking all the writes and process starts/stops.  Let's do what's practical with the existing API object and stop there.